src/wrappers/sheltering_transfers_wrapper.js

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright: Ambrosus Inc.
+Email: tech@ambrosus.com
+
+This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+This Source Code Form is “Incompatible With Secondary Licenses”, as defined by the Mozilla Public License, v. 2.0.
+*/
+
+import {MIN_BLOCK_TIME} from '../constants';
+import ManagedContractWrapper from './managed_contract_wrapper';
+
+export default class ShelteringTransfersWrapper extends ManagedContractWrapper {
+  get getContractName() {
+    return 'shelteringTransfers';
+  }
+
+  async earliestMeaningfulBlock(transferDuration) {
```

Comment:
Theoretically, a situation is possible when for TransferStarted there is no corresponding TransferResolved or TransferCancelled. If we do not limit the search for events in the blockchain, then at the start of Atlas we will find these unanswered requests. To prevent this, I propose to limit the search for events in the blockchain to the challenge duration.

Reply:
Still I can't see any reason this should be in the wrapper.
The Atlas worker itself can employ any sensible strategy for limiting the search.